src/app/modules/main-module/components/main/main.component.ts

```diff
@@ -10,50 +10,98 @@ import { DialogComponent } from 'src/app/components/dialog/dialog.component';
   templateUrl: './main.component.html',
   styleUrls: ['./main.component.scss'],
 })
-export class MainComponent implements OnInit{
+export class MainComponent implements OnInit {
   palabraModel: Palabra = {
-    nombre: '',
+    pos0: '',
+    pos1: '',
+    pos2: '',
+    pos3: '',
+    pos4: '',
   };
 
-  disable: boolean=false;
+  word = Object.values(this.palabraModel);
+
+  posicionInput = 0;
+  wordSend = '';
+  disableKeyboard: boolean = false;
   teclado: string[] = TECLADO;
   inicioPalabra: number = 0;
   finPalabra: number = -1;
 
   constructor(private gameService: GameService, private dialog: MatDialog) {}
 
-  ngOnInit(){
+  ngOnInit() {
     this.gameService.getDisable().subscribe({
-      next: (response:any) => {
-        this.disable= response
-      }
-    })
+      next: (response: any) => {
+        this.disableKeyboard = response;
+      },
+    });
   }
 
   sendWord() {
-    this.gameService.getWordIfExist(this.palabraModel.nombre).subscribe({
-      next: (response:any) => {
-        if (response.wordExists) return;
-        this.openDialog();
+    this.gameService.getWordIfExist(this.buildWord()).subscribe({
+      next: (response: any) => {
+        if (response.wordExists) {
```

Comment:
aquí hay que refactorizar, esta función hace demasiadas cosas:
paramezar utilizar el guarda de manera que si no hay wordExist llame a this.openDialog(); y return, que no continue.
getValidatePosition debe ir a otra función...crear una función para realizar esta lógica